packages/ui-app/src/TxButton.tsx

```diff
@@ -29,16 +34,59 @@ type Props = ApiProps & {
   onSuccess?: TxCallback,
   onUpdate?: TxCallback,
   params?: Array<any> | ConstructFn,
-  tx: string
+  tx: string | IExtrinsic | SubmittableExtrinsic
 };
 
-class TxButtonInner extends React.PureComponent<Props & InjectedProps> {
+type InnerProps = Props & InjectedProps;
+
+type State = {
+  extrinsic: SubmittableExtrinsic,
+  isSending: boolean
+};
+
+class TxButtonInner extends React.PureComponent<InnerProps> {
+  state = {
+    isSending: false
+  } as State;
+
+  static getDerivedStateFromProps ({ api, params = [], txqueue = [], tx = '' }: InnerProps): State | null {
+    if (!tx || !(tx as string).length) {
+      return null;
+    }
+
+    let extrinsic: any;
+    if (typeof tx === 'string') {
+      const [section, method] = tx.split('.');
+
+      assert(api.tx[section] && api.tx[section][method], `Unable to find api.tx.${section}.${method}`);
+
+      extrinsic = api.tx[section][method](
+        ...(isFunction(params) ? params() : params)
+      );
+    } else {
+      extrinsic = tx as SubmittableExtrinsic;
```

Comment:
Not 100% sure about this - this is used in extrinsic app and that one is a bit special, and probably the only one with any reason to pull in the queue itself. Maybe rather make it explicit and pass in an (optional) `extrinsic` prop?

Reply:
Done - extrinsic either passed or generated from tx string